File: hw/hw.go
package main

import (
    "net/http"
    "strconv"
    "unit.nginx.org/go"
)

type def struct {
    length     int
    length_str string
}

func make_def(l int) *def {
    return &def{l, strconv.Itoa(l)}
}

var body_huge []byte
var body_def map[string]*def
var body_14 *def

func handler(w http.ResponseWriter, r *http.Request) {
    d, found := body_def[r.RequestURI]

    if !found {
        d = body_14
    }

    w.Header().Add("Content-Type", "text/plain")
    w.Header().Add("Content-Length", d.length_str)

    w.Write(body_huge[:d.length])
}

func main() {
    body_huge = make([]byte, 1024 * 1024)
    body := []byte("Hello, world!\n")

    for i := range body_huge {
        body_huge[i] = body[i % len(body)]
    }

    body_14 = make_def(len(body))

    body_def = map[string]*def {
        "/": body_14,
        "/1k": make_def(1024),
        "/4k": make_def(1024 * 4),
        "/16k": make_def(1024 * 16),
        "/64k": make_def(1024 * 64),
        "/1m": make_def(1024 * 1024),
    }

    http.HandleFunc("/", handler)
    unit.ListenAndServe(":8002", nil)
}

File: hw/hw.js
#!/usr/bin/env node

if (process.env.NXT_UNIT_INIT) {
  http = require('unit-http')
} else {
  http = require('http')
}

const body_1m = Buffer.alloc(1024 * 1024, 'Hello, world!\n');
const body_14 = body_1m.slice(0, 14);
const body_def = {
    '/':    body_1m.slice(0, 14),
    '/1k':  body_1m.slice(0, 1024),
    '/4k':  body_1m.slice(0, 1024 * 4),
    '/16k': body_1m.slice(0, 1024 * 16),
    '/64k': body_1m.slice(0, 1024 * 64),
    '/1m':  body_1m.slice(0, 1024 * 1024),
};

http.createServer(function (req, res) {
    let body = body_def[req.url] || body_14;

    res.setHeader('Content-Length', body.length);
    res.setHeader('Content-Type', 'text/plain');
    res.writeHead(200, {}).end(body);
}).listen(8003);
